planet_explorer/gui/pe_orders_monitor_dockwidget.py

```diff
@@ -285,6 +310,161 @@ def download(self, is_unit_test=False):
+        a grey scale layer, whereas multiband will be added as True colour RGB.
+
+        :param layer: Raster layer. Both single band and multiband.
+        :type layer: QgsRasterLayer
+        """
+
+        band_cnt = layer.bandCount()
+        if band_cnt < 3:
+
+            # These cases will be skipped for now, but removing this 'return'
+            # will add singleband layers again
+            return
+
+            # Rasters with less than 3 bands will be added as single band
+            r = layer.renderer().clone()
+            r.setGrayBand(1)
+
+            used_bands = r.usesBands()
+            typ = layer.renderer().dataType(1)
+            enhancement = QgsContrastEnhancement(typ)
+            enhancement.setContrastEnhancementAlgorithm(
+                QgsContrastEnhancement.StretchToMinimumMaximum, True
+            )
+            band_min, band_max = layer.dataProvider().cumulativeCut(
+                used_bands[0], 0.02, 0.98, sampleSize=10000
+            )
+            enhancement.setMinimumValue(band_min)
+            enhancement.setMaximumValue(band_max)
+            r.setContrastEnhancement(enhancement)
+
+            layer.setRenderer(r)
+            QgsProject.instance().addMapLayer(layer)
+        else:
+            # BGR image for 3 or more bands
+            r = layer.renderer().clone()
+            r.setBlueBand(self._find_band(layer, "blue", 1))
+            r.setGreenBand(self._find_band(layer, "green", 2))
+            r.setRedBand(self._find_band(layer, "red", 3))
+
+            used_bands = r.usesBands()
+            for b in range(3):
+                typ = layer.renderer().dataType(b)
+                enhancement = QgsContrastEnhancement(typ)
+                enhancement.setContrastEnhancementAlgorithm(
+                    QgsContrastEnhancement.StretchToMinimumMaximum, True
+                )
+                band_min, band_max = layer.dataProvider().cumulativeCut(
+                    used_bands[b], 0.02, 0.98, sampleSize=10000
+                )
+                enhancement.setMinimumValue(band_min)
+                enhancement.setMaximumValue(band_max)
+                if b == 0:
+                    r.setRedContrastEnhancement(enhancement)
+                elif b == 1:
+                    r.setGreenContrastEnhancement(enhancement)
+                elif b == 2:
+                    r.setBlueContrastEnhancement(enhancement)
+
+            layer.setRenderer(r)
+            QgsProject.instance().addMapLayer(layer)
+
+    def add_to_map(self):
+        """Called when the add to map button is clicked.
+        Adds the selected remotely sensed image in the order monitor list to QGIS.
+        The data needs to be downloaded.
+        """
+
+        # Order name is usually "OrderName_" followed by the sensor (e.g. SkySat)
+        # For the QGIS plugin the output folder should be "OrderName_QGIS"
+        order_name_split = self.order.name().split("_")
```

Comment:
This is only going to work for orders created via the QGIS plugin. We should support orders that were created from a source other than the QGIS plugin too.

Reply:
@john-dupuy I were not aware there should be other sources included. What other sources other than the QGIS plugin does exist? ArcMap? If that is the case, there is no way for me to test it.

Reply:
Orders can be created in numerous ways: https://developers.planet.com/apis/orders/

Most commonly orders will be created via the planet.com/explorer and those are allowed to have fairly arbitrary names. I expect customers will often order things via explorer and then analyze them in QGIS, so we should support adding these orders onto the map.